### DISCUSSION TITLE
Any pracitcal example how to use async file system API for reading files?

### DISCUSSION THREAD

**Discussion by sasmaster** [emscripten-core/emscripten#18161] [Q&A] [answered]
I am not sure I understand how to use `emscripten_async_wget_data` for reading files from the disk.
I open a file dialog via <input type="file"> and trying to fetch the URL into emscripten_async_wget_data to read it in my WebAssembly app. It doesn't work. Anyone can point to an example?
Here is the complete problem described in[ this SO post](https://stackoverflow.com/questions/74351374/loading-files-from-disk-using-emscripten-async-file-system).

Thanks.

**Comment by kripken** [comment, marked as answer]
There is a test in the test suite here, that might be useful:

https://github.com/emscripten-core/emscripten/blob/7d291a7660cbf678dac9988dda7b898570ba4ed0/test/emscripten_fs_api_browser.c#L53-L57

**Reply by sasmaster** [reply]
Yes, I see.Actually I was looking for an example how to load file from local file system after selecting it in a file explorer dialog. Then I found
that URL.createObjectURL  works as well.
